src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = new Node<>(null, value, null);
+        } else if (tail == null) {
+            tail = new Node<>(head, value, null);
+            head.next = tail;
+        } else {
+            Node<T> element = new Node<>(tail, value, null);
+            tail.next = element;
+            tail = element;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            head = new Node<>(null, value, getNode(index));
+            getNode(index).prev = head;
+            size++;
+        } else {
+            Node<T> previous = getNode(index).prev;
+            Node<T> newNode = new Node<>(previous, value, getNode(index));
+            previous.next = newNode;
+            getNode(index).prev = newNode;
```

Comment:
you need to create separate variable for getNode(index) and use this variable in your statements